### DISCUSSION TITLE
Help with some data integration

### DISCUSSION THREAD

**Discussion by dominikkubik** [Budibase/budibase#12695] [Help] [answered]
Hi.
I already searched this discussion forum as well as documentation, but i still need some help.
What i try to run up is application with two screens(input forms) that will utilize two external data sources.
Use case scenario explained simply is:
1. Application shows 1.screen and user input all field.
2. User click the Save button.
3. Application validates the data
4. Application saves data into PostgreSQL database
5. Application runs POST query to save the same data into external system.
6. Application should get the row ID of saved data from PostgreSQL as well as ID returned by POST request.
7. Application shows 2.screen where PostgreSQL DB ID and External ID(returned as reply by POST) are also displayed
8. User input all other data
9. User click the Save button
10. Application validate the data in form.
11. Application saved data into PostgreSQL DB and also via POST into external system.

And what i am still missing and I can't find how to do that is actions in step 6: 6. Application should get the row ID of saved data from PostgreSQL as well as ID returned by POST request.

I dont know how to get, store and use in scenario the data, ID of row in DB as well as ID returned by POST reply.
I tried to set up variables, bindings etc. but I am still not able to get and use those ID's.

Any idea how or tip please?

EDIT: I also tried to save the results received from a post API request to state using the actions bindings as mentioned in https://github.com/Budibase/budibase/discussions/11976  but still not working

Kind regards
Dominik

PF 2024 !!!

**Comment by melohagan** [comment, marked as answer]
Hey @dominikkubik 

Have you had a look at this page of the docs? https://docs.budibase.com/docs/chaining-actions

You can access the IDs using bindings. For example the saved row: `{{ Action 4.Saved Row._id }}` and the ID from the POST query via: `{{ Action X.Query Result.data.0.id }}`

**Comment by dominikkubik** [comment]
Hey @melohagan 

No I missed the topic of Chaining actions and I don't know why :(
Thank you very much i will read it and definitely try it.

Thank's a lot.

Kind regards
Dominik

**Comment by dominikkubik** [comment]
> You can access the IDs using bindings. For example the saved row: {{ Action 4.Saved Row._id }} and the ID from the POST query via: `{{ Action X.Query Result.data.0.id }}

This is only possible when accessing action in the same screen/action panel. I did not find the way to access it from different screen.
What i am trying to use is to update states and get values from this states.

But still not working.

**Reply by melohagan** [reply]
You may need to persist the state to access it across screens. Alternatively you could try using URL variables: https://docs.budibase.com/docs/url-parameters
